### ISSUE TITLE
z.discriminatedUnion and z.string.array throw SchemaError when submitting

### ISSUE THREAD

**Issue by endygower** [ciscoheat/sveltekit-superforms#516]
- [x] Before posting an issue, read the [FAQ](https://superforms.rocks/faq) and search the previous issues.

**Description**
while I read that discriminatedUnion might be deprecated  ([here](https://github.com/ciscoheat/sveltekit-superforms/issues/231#issuecomment-1637123932)). however, this issue also reproducible using `valibot.variant`. I don't know whether valibot might be deprecated as well

I just so happen to write this using zod

Using `z.discriminatedUnion` and `z.string().array()` ie:
```js
const schema = z.discriminatedUnion('type', [
	z.object({ type: z.literal('empty') }),
	z.object({ type: z.literal('extra'), options: z.string().array() })
])
```
it throws `SchemaError` on the server

However, it works when using `z.string()`
```js
const schema1 = z.discriminatedUnion('type', [
	z.object({ type: z.literal('empty') }),
	z.object({ type: z.literal('extra'), options: z.string() })
])
```

**Investigation**
I did some investigation and found the responsible code is in this line
https://github.com/ciscoheat/sveltekit-superforms/blob/f2201d11c4c2e9b8553b69a137ab246586c87a16/src/lib/formData.ts#L201-L209
when parsing the form data, `property` is set as `defaultPropertyType` which is a `{ type: 'string' }`

it doesn't consider that on top level union the schema is as follow and doesn't have `properties`
```js
{
  $schema: ...,
  anyOf: [...]
}
```

then it will call `parseSingleValue` and the output will be
```js
const data = new FormData()
data.append('options', '1')
data.append('options', '2')

imaginaryCallTo_parseFormData(data, ...)
// actual: { options: '2' }
// expected: { options: ['1', '2'] }
```

which will always fail the zod schema and later throws `SchemaError`

**If applicable, a MRE**
inside +page.server.ts you can change the schema use between `schema` (z.string.array) and `schema1` (z.string)
https://www.sveltelab.dev/4adylfnc6g7pv98

**Comment by ciscoheat** [comment]
Thank you for a very thorough investigation, will hopefully fix this for the next release.

**Comment by ciscoheat** [comment]
Can you just confirm that the same thing happens when dataType is set to 'json' (or not) for the form?

**Comment by endygower** [comment]
I just checked it sends the form correctly using dataType `json`.
I also updated the MRE if you want to try it yourself

Meanwhile, I will use dataType `json` while it is being fixed. thanks @ciscoheat

**Comment by ciscoheat** [comment]
Ok, then the "fix" will be to issue an error, since unions aren't allowed unless dataType is set to json.

**Comment by ciscoheat** [comment]
I think this will be properly fixed in the next release, as #519 had a similar problem.
